services/pull/update.go

```diff
@@ -34,17 +34,20 @@ func Update(ctx context.Context, pr *issues_model.PullRequest, doer *user_model.
 	}
 	defer releaser()
 
-	diffCount, err := GetDiverging(ctx, pr)
-	if err != nil {
-		return err
-	} else if diffCount.Behind == 0 {
-		return fmt.Errorf("HeadBranch of PR %d is up to date", pr.Index)
-	}
-
 	if err := pr.LoadBaseRepo(ctx); err != nil {
 		log.Error("unable to load BaseRepo for %-v during update-by-merge: %v", pr, err)
 		return fmt.Errorf("unable to load BaseRepo for PR[%d] during update-by-merge: %w", pr.ID, err)
 	}
+
+	if pr.ID > 0 { // only a real PR needs to check this FIXME: why it needs this check?
```

Comment:
FIXME

Reply:
The `MergeUpstream` created a virtual pull request which have an false index which is -1. But `GetDiverging` need a valid index number. And it seems both the base branch and head branch are the same. So that we don't need to invoke GetDiverging.